src/System.Private.TypeLoader/src/Internal/Runtime/TypeLoader/DebugFuncEval.cs

```diff
@@ -0,0 +1,143 @@
+    }
+
+    internal class AutoPinner : IDisposable
+    {
+        private GCHandle _pinnedArray;
+
+        public AutoPinner(Object obj)
+        {
+            _pinnedArray = GCHandle.Alloc(obj, GCHandleType.Pinned);
+        }
+
+        public static implicit operator IntPtr(AutoPinner ap)
+        {
+            return ap._pinnedArray.AddrOfPinnedObject();
+        }
+        
+        public void Dispose()
+        {
+            _pinnedArray.Free();
+        }
+    }
+}
+
+namespace Internal.Runtime.TypeLoader
+{
+    internal class DebugFuncEval
+    {
+
+        private static void HighLevelDebugFuncEvalHelperWithVariables(ref int param, ref LocalVariableSet arguments)
+        {
+            // Hard coding the argument integer here!
+            arguments.SetVar<int>(1, param);
+
+            // For now, fill this variable in the debugger, any static void method with a single integer argument would work
+            IntPtr targetAddress = RuntimeImports.RhpGetFuncEvalTargetAddress();
+
+            // Hard coding a single void return here
+            LocalVariableType[] returnAndArgumentTypes = new LocalVariableType[2];
+            returnAndArgumentTypes[0] = new LocalVariableType(typeof(void).TypeHandle, false, false);
+            returnAndArgumentTypes[1] = new LocalVariableType(typeof(int).TypeHandle, false, false);
+
+            // 2) Invoke the target method
+
+            // Hard coding static here
+            DynamicCallSignature dynamicCallSignature = new DynamicCallSignature(Internal.Runtime.CallConverter.CallingConvention.ManagedStatic, returnAndArgumentTypes, returnAndArgumentTypes.Length);
+
+            // Go and execute!
+            Internal.Runtime.CallInterceptor.CallInterceptor.MakeDynamicCall(targetAddress, dynamicCallSignature, arguments);
+
+            FuncEvalCompleteCommand funcEvalCompleteCommand = new FuncEvalCompleteCommand();
+            funcEvalCompleteCommand.commandCode = 0;
+
+            using (AutoPinner ap = new AutoPinner(funcEvalCompleteCommand))
+            {
+                RuntimeImports.RhpSendCustomEventToDebugger(ap, Unsafe.SizeOf<FuncEvalCompleteCommand>());
+            }
+        }
+
+        [StructLayout(LayoutKind.Explicit, Size=16)]
+        struct WriteParameterCommand
+        {
+            [FieldOffset(0)]
+            public int commandCode;
+            [FieldOffset(4)]
+            public int unused;
+            [FieldOffset(8)]
+            public long bufferAddress;
+        }
+
+        [StructLayout(LayoutKind.Explicit, Size=4)]
+        struct FuncEvalCompleteCommand
+        {
+            [FieldOffset(0)]
+            public int commandCode;
+        }
+
+        private static void HighLevelDebugFuncEvalHelper()
+        {
+            int integerParameterValue = 0;
+
+            uint parameterBufferSize = RuntimeImports.RhpGetFuncEvalParameterBufferSize();
+            byte[] debuggerBuffer = new byte[parameterBufferSize];
```

Comment:
How big can this buffer size get? Would it make sense to `stackalloc` and avoid pinning?